New Query Enhancement UI

abbyhu2000 opened the pull request:
### Description

New query enhancement UI. 

### Issues Resolved

#7038 

## Screenshot


https://github.com/user-attachments/assets/c6079810-d38e-4b6e-b272-d16e5ddbdf5f



## Testing the changes

<!--
  Please provide detailed steps for validating your changes. This could involve specific commands to run,
  pages to visit, scenarios to try or any other information that would help reviewers verify
  the functionality of your change
-->

## Changelog

- feat: Update query enhancement UI


### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector.tsx:
```diff
@@ -3,16 +3,25 @@
   anchorPosition?: PopoverAnchorPosition;
   appName?: string;
+  languageSelectorContainerRef?: React.RefCallback<HTMLDivElement>;
+  noArrow?: boolean;
```
nit: i think we should just get rid of the arrow all together right?

abbyhu2000 replied:
removed

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector.tsx:
```diff
@@ -22,7 +31,18 @@ const mapExternalLanguageToOptions = (language: string) => {
+  };
+
+  const closePopover = () => {
+    setPopover(false);
```
nit: do we need these? these are just one line anyways

abbyhu2000 replied:
agreed, made it one liner to reduce code

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector.tsx:
```diff
@@ -64,25 +84,56 @@ export const QueryLanguageSelector = (props: Props) => {
-    const queryLanguage = newLanguage[0].value as string;
-    props.onSelectLanguage(queryLanguage);
-    uiService.Settings.setUserQueryLanguage(queryLanguage);
+  const handleLanguageChange = (newLanguage: any) => {
```
how come the change to any?

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector.tsx:
```diff
@@ -64,25 +84,56 @@ export const QueryLanguageSelector = (props: Props) => {
+    uiService.Settings.setUserQueryLanguage(newLanguage);
   };
 
-  uiService.Settings.setUserQueryLanguage(props.language);
```
this was here to ensure the initial state has the correct language the first time. likely this might be causing a bug where if you select another language like PPL. and reload the page the actually query language is not the right value so it fails the search request

abbyhu2000 replied:
i selected PPL, and then after refreshing and reloading the page, the query language still remains PPL. Were there other scenarios that make this line necessary?

abbyhu2000 replied:
added the line back

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector.tsx:
```diff
@@ -64,25 +84,56 @@ export const QueryLanguageSelector = (props: Props) => {
+    </EuiButtonEmpty>
+  );
+  const getIconType = (language: string) => {
+    return language === selectedLanguage.label ? 'check' : 'empty';
```
how come? because it's buggy? or we want it to have the check?

abbyhu2000 replied:
when user clicks the language selector button, this is rendering the check icon next to whatever language being selected. 
![Screenshot 2024-07-19 at 1 58 43 AM](https://github.com/user-attachments/assets/3a114d6c-6c47-4125-8f2e-e8692a58a20e)

abbyhu2000 replied:
also made it a one-liner instead of a new function

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector.tsx:
```diff
@@ -64,25 +84,56 @@ export const QueryLanguageSelector = (props: Props) => {
 
+  const languageOptionsMenu = languageOptions.map((language) => {
+    return (
+      <EuiContextMenuItem
```
oh you are making it a context menu now. i see. i'm good with that.

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector_footer_popover.tsx:
```diff
@@ -0,0 +1,144 @@
+  };
+
+  // language selector at the footer of the query bar
+  const button = (
```
how come const

is this re-used somewhere? if not then we should just use it inline

abbyhu2000 replied:
made the button inline instead of a const

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector_footer_popover.tsx:
```diff
@@ -0,0 +1,144 @@
+      iconSide="right"
+      onClick={onButtonClick}
+      style={{
+        fontSize: 'smaller',
```
should make this a class and then apply.

and based on the look and feel project. we should respect the OUI defined guides for styling this button

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector.tsx:
```diff
@@ -64,25 +84,56 @@ export const QueryLanguageSelector = (props: Props) => {
+    return (
+      <EuiContextMenuItem
+        key={language.label}
+        icon={getIconType(language.label)}
```
can maybe do a 1 line function for this instead of making another function where we checked it here

kavilla commented on src/plugins/data/public/ui/query_editor/language_selector_footer_popover.tsx:
```diff
@@ -0,0 +1,144 @@
+      {props.language}
+    </EuiButtonEmpty>
+  );
+  const getIconType = (language: string) => {
```
oh this looks redundant of the same component function. we shouldn't need to have two components if just doing styling changes or other things that can be modified.

kavilla replied:
that way if there is a bug in one spot then there isn't a bug in the other spot still

abbyhu2000 replied:
deleted this file, and only use the language selector component

kavilla commented on src/plugins/data/public/ui/query_editor/query_editor.tsx:
```diff
@@ -85,9 +109,12 @@ export default class QueryEditorUI extends Component<Props, State> {
     index: null,
     suggestions: [],
     indexPatterns: [],
+    isCollapsed: true,
```
should be false at first right?

abbyhu2000 replied:
fixed the naming. The default mode is expanding

ashwin-pc reviewed (commented):
The styling still needs some fixes:

<img width="1108" alt="image" src="https://github.com/user-attachments/assets/1b7e6b02-893e-4bfd-bdad-facea937780e">

1. There is a small padding between the query bar and the side nav thats making the query bar look odd
2. The footer does not use OUI theme colors
3. Why is the default language PPL?

For DQL:

<img width="652" alt="image" src="https://github.com/user-attachments/assets/220ef444-0fd5-42a3-91ac-0b656f10825f">


1. The language selector border is not using OUI colors


And when the toggle is off:
<img width="216" alt="image" src="https://github.com/user-attachments/assets/f24445d9-9def-4ee4-b209-879ddc106b39">


1. the save icon is missing the dropdown arrow
2. The filter icon is missing the horizontal line next to it

abbyhu2000 commented:
@ashwin-pc 

1. fix the padding
2. changed footer to use OUI theme color
3. the default is not PPL, it is based on local storage
4. changed the language shadow box to use OUI theme color
5. based on discussion @kgcreative we agree to get rid of the arrow on the right of the saved query management

ashwin-pc reviewed (approved):
Nice, this gets the framework in. Since we have gone the custom route, lets clean up the styles to make the fit and finish better.
